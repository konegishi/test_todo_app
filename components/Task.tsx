import React from 'react';

/**
 * TaskのProps
 */
export interface TaskProps {
  /** タスクのタイトル */
  title: string;
}

/**
 * Taskコンポーネント
 * @param props TaskのProps
 * @returns Taskコンポーネント
 */
const Task: React.FC<TaskProps> = (props) => {
  return (
    <React.Fragment>
      <div className='relative bg-white rounded-xl mb-4 shadow-lg p-2'>
        <div className='flex flex-row items-center'>
          <input
            type='checkbox'
            className='h-6 w-6 rounded m-2 border-gray-300 focus:border-indigo-300 focus:ring-2 focus:ring-indigo-200 focus:ring-opacity-50 text-indigo-500'
          />
          <span className='text-xl text-blueGray-700'>{props.title}</span>
          <button className='fas fa-ellipsis-v text-gray-500 absolute right-4 ' />
        </div>
      </div>
    </React.Fragment>
  );
};

export default Task;
