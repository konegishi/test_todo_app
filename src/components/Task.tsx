import React, { useState } from 'react';
// import { useTodos } from '../hooks/useTodos';
import TaskModal from './TaskModal';

/**
 * TaskのProps
 */
export interface TaskProps {
  /** タスク名 */
  name: string;
  /** タスクのid */
  id: number;
}

/**
 * Taskコンポーネント
 * 1タスクを表すコンポーネント
 *
 * @param props TaskのProps
 * @returns Taskコンポーネント
 */
const Task: React.FC<TaskProps> = (props) => {
  const [isOpen, setOpen] = useState(false);
  // const { deleteTodo } = useTodos();

  return (
    <React.Fragment>
      <div className='flex flex-row items-center border-t border-gray-200 mb-2 pt-2'>
        <input
          type='checkbox'
          className='h-5 w-5 form-checkbox rounded-full mx-2 border-gray-300 focus:border-indigo-300 focus:ring-2 focus:ring-indigo-200 focus:ring-opacity-50 text-indigo-500'
        />
        <span className='text-xl text-blueGray-700 ml-2'>{props.name}</span>
        <button
          className='fas fa-chevron-right h-5 w-5 text-gray-400 ml-auto mr-2'
          onClick={() => setOpen(!isOpen)}
        />
        {/* <i class="fas fa-chevron-right"></i> */}
        {/* <button
          className='far fa-trash-alt h-5 w-5 text-gray-500 ml-auto mr-2'
          onClick={() => deleteTodo(props.id)}
        /> */}
      </div>
      {isOpen && <TaskModal onClickHandler={setOpen} />}
    </React.Fragment>
  );
};

export default Task;
