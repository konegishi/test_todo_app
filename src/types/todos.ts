import { User } from '@supabase/supabase-js';

/**
 * Todoの型定義
 */
export interface Todo {
  /** todo名 */
  task: string;
  /** todoのid */
  id: number;
  /** 説明 */
  description: string;
  /** 完了フラグ */
  isComplete: boolean;
}

export interface UseTodos {
  /** todoのリスト */
  todos: Todo[];
  /** ローディング中フラグ */
  isLoading: boolean;
  /**addTodoメソッドの型定義 */
  addTodo: (user: User, taskText: string) => Promise<void>;
  /** updateTodoメソッドの型定義 */
  updateTodo: (
    id: number,
    todo: {
      task?: string;
      description?: string;
      isComplete?: boolean;
    }
  ) => Promise<void>;
  /** deleteTodoメソッドの型定義 */
  deleteTodo: (id: number) => Promise<void>;
}
