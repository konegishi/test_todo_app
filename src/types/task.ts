import { UseTodos } from './todos';

/**
 * TaskコンポーネントのProps
 */
export interface TaskProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** タスクの完了フラグ */
  isComplete: boolean;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
