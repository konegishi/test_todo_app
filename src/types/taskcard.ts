import { Todo, UseTodos } from './todos';

/**
 * TaskCardコンポーネントのProps
 */
export interface TaskCardProps {
  /** タスクのリスト */
  todos: Todo[];
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
