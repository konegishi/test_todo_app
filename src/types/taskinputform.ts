import { User } from '@supabase/supabase-js';
import { UseTodos } from './todos';

/**
 * TaskInputFormコンポーネントのProps
 */
export interface TaskInputFormProps {
  /** ユーザー情報 */
  user: User;
  /** タスク名 */
  name?: string;
  /** AddボタンのHandler */
  addTodoHandler: UseTodos['addTodo'];
}
