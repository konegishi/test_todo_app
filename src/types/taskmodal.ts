import { Dispatch, SetStateAction } from 'react';
import { UseTodos } from './todos';

/**
 * TaskModalコンポーネントのProps
 */
export interface TaskModalProps {
  /** タスク名 */
  title: string;
  /** タスクのid */
  id: number;
  /** タスクの説明 */
  description: string;
  /** Modal開閉用メソッド */
  onClickHandler: Dispatch<SetStateAction<boolean>>;
  /** DeleteボタンのHandler */
  deleteTodoHandler: UseTodos['deleteTodo'];
  /** UpdateボタンのHandler */
  updateTodoHandler: UseTodos['updateTodo'];
}
