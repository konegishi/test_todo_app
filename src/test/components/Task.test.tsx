import { render } from '@testing-library/react';
import React from 'react';
import Task from '../../components/Task';
import { TaskProps } from '../../types/task';

describe('Task component', () => {
  it('matches snapshot', () => {
    const dummyProps: TaskProps = {
      title: '型定義を追加する',
      id: 10,
      description: 'types以下に配置する',
      isComplete: false,
      deleteTodoHandler: jest.fn(),
      updateTodoHandler: jest.fn(),
    };
    const { asFragment } = render(<Task {...dummyProps} />, {});
    expect(asFragment()).toMatchSnapshot();
  });
});
