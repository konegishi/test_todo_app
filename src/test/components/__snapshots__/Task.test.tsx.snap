// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Task component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-row items-center border-t border-gray-200 py-2"
  >
    <input
      class="h-5 w-5 form-checkbox rounded-full mx-2 border-gray-300 focus:ring-0 focus:ring-opacity-50 text-blue-500"
      type="checkbox"
    />
    <span
      class="text-lg ml-2 truncate text-blueGray-800"
    >
      型定義を追加する
    </span>
    <button
      class="fas fa-ellipsis-v h-5 w-5 ml-auto mr-2 text-gray-400"
    />
  </div>
</DocumentFragment>
`;
